crates/kv-store/src/block.rs

```diff
@@ -0,0 +1,623 @@
+use std::{
+    fmt::Debug,
+    ops::{Bound, Range},
+    sync::Arc,
+};
+
+use bytes::{Buf, BufMut, Bytes, BytesMut};
+
+use crate::{iter::KvIterator, sstable::{get_common_prefix_len_and_strip, SIZE_OF_U32, XXH_SEED}};
+
+use super::sstable::{ SIZE_OF_U16, SIZE_OF_U8};
+
+
+#[derive(Debug)]
+pub struct LargeValueBlock{
+    // without checksum
+    pub value_bytes: Bytes,
+    pub key: Bytes,
+}
+
+impl LargeValueBlock{
+    /// ┌──────────────────────────┐
+    /// │Large Block               │
+    /// │┌ ─ ─ ─ ┬ ─ ─ ─ ─ ─ ─ ─ ─ │
+    /// │  value   Block Checksum ││
+    /// ││ bytes │      u32        │
+    /// │ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘│
+    /// └──────────────────────────┘
+    fn encode(&self)->Bytes{
+        let mut buf = Vec::with_capacity(self.value_bytes.len() + SIZE_OF_U32);
+        buf.put_slice(&self.value_bytes);
+        let checksum = xxhash_rust::xxh32::xxh32(&buf, XXH_SEED);
+        buf.put_u32(checksum);
+        buf.into()
+    }
+
+    fn decode(bytes:Bytes, key: Bytes)->Self{
+        LargeValueBlock{
+            value_bytes: bytes.slice(..bytes.len() - SIZE_OF_U32),
+            key,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct NormalBlock {
+    pub data: Bytes,
+    pub first_key: Bytes,
+    pub offsets: Vec<u16>,
+}
+
+impl NormalBlock {
+    /// ┌────────────────────────────────────────────────────────────────────────────────────────┐
+    /// │Block                                                                                   │
+    /// │┌ ─ ─ ─ ─ ─ ─ ─ ┬ ─ ─ ─┌ ─ ─ ─ ─ ─ ─ ─ ┬ ─ ─ ─ ─┌ ─ ─ ─┌ ─ ─ ─ ┬ ─ ─ ─ ─┌ ─ ─ ─ ─ ─ ─ ─ │
+    /// │ Key Value Chunk  ...  │Key Value Chunk  offset │ ...  │ offset  kv len │Block Checksum││
+    /// ││     bytes     │      │     bytes     │  u16   │      │  u16  │  u16   │     u32       │
+    /// │ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ┘│
+    /// └────────────────────────────────────────────────────────────────────────────────────────┘
+    /// 
+    /// check sum will be calculated by crc32 later
+    fn encode(&self) -> Bytes {
+        let mut buf = self.data.to_vec();
+        for offset in &self.offsets {
+            buf.put_u16(*offset);
+        }
+        buf.put_u16(self.offsets.len() as u16);
+
+        let mut compressed_data = lz4_flex::compress_prepend_size(&buf);
```

Comment:
I want the used compression to be configurable in the future. And we can have a field in meta that indicates how to compress the data (this flag can share the same byte with `is_large`)